src/shortest_path/floyd_warshall.rs

```diff
@@ -149,11 +149,18 @@ pub fn floyd_warshall_numpy<Ty: EdgeType>(
     as_undirected: bool,
     default_weight: f64,
     parallel_threshold: usize,
-) -> PyResult<Array2<f64>> {
+) -> PyResult<(Array2<f64>, Array2<usize>)> {
```

Comment:
Can you duplicate this code? I think the overall implementation is fine but I don't want to add the extra overhead of keeping the predecessors for the existing `floyd_warshall_numpy`